### PR TITLE
test: Added unit tests for Available Batch Report covering filters, expiry handling, and batch-wise stock data

### PR DESCRIPTION
**Pull request by ashishverma8848**
### Title:
Test Case's for Available Batch Report in ERPNext

### Test Summary:
This test suite provides comprehensive unit test coverage for the ERPNext report available_batch_report.py. It validates the report's functionality under various filtering and data conditions, ensuring the correctness of batch-wise inventory reporting, proper handling of edge cases, and structural integrity of output data (columns and values).

### Scenarios Covered:
1.Data Setup
  Creation of company (_Test Company)
  Creation of warehouse (Stores - W1 - _TC)
  Item creation (TEST-ITEM-100)
  Batch creation (BATCH-001)
  Stock entry of item with batch and warehouse
  Fiscal year setup

2.Data Fetching & Logic Tests
  No data scenario: empty results when item or filters don't match
  Valid filters returning expected structured output
  Batch expiry filtering with:
  to_date filter
  include_expired_batches flag
  expiry_date filter
  Warehouse filtering via:
  warehouse name
  warehouse_type
  Batch number filtering
  Toggle on show_item_name
  Multiple filters applied simultaneously

3.Function-specific Testing
  get_data(filters) with different combinations of filters
  get_columns(filters) with show_item_name=True/False
  execute(filters) verifying return of both columns and data

### Technology:
Framework: Frappe / ERPNext
Language: Python
Test Utility: FrappeTestCase (Frappe's unit testing base class)

### Modules Used:
frappe
erpnext.stock.report.available_batch_report
frappe.utils.today
types.SimpleNamespace (for dynamic filters)
Test Strategy: Unit tests using in-memory data setup and verification

### Test Case Ids:
  test_get_batchwise_data
  test_get_batchwise_data_empty
  test_get_batchwise_data_zero_qty
  test_get_batchwise_data_none_qty
  test_get_batchwise_data_no_filters
  test_get_batchwise_data_expiry_date
  test_get_batchwise_data_warehouse
  test_get_batchwise_data_batch_no
  test_get_batchwise_data_show_item_name
  test_get_batchwise_data_include_expired_batches
  test_get_batchwise_data_to_date
  test_get_batchwise_data_multiple_filters
  test_execute_function
  test_get_columns_with_item_name
  test_get_columns_without_item_name
  test_to_date_today_with_expiry_check
  test_to_date_today_with_expired_batches_included
  test_filter_by_warehouse_type

### CODE REVIEW

**Comment by aasif-patel** [comment]
@ashishverma8848 please update the title

**Comment by aasif-patel** [comment]
@ashishverma8848 Test case id's are missing, do mention those in PR description as well
CC @pallavi238848

**Comment by ashishverma8848** [comment]
@aasif-patel updated the test case ids

File: erpnext/stock/report/available_batch_report/test_available_batch_report.py [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+			batch_no=None,
+			warehouse_type=None,
+		)
+		default_filters.update(overrides)
+		return SimpleNamespace(**default_filters)
+
+	def test_get_batchwise_data(self):
+		filters = self.make_filters(item_code=self.item.name)
+		data = available_batch_report.get_data(filters)
+		self.assertEqual(len(data), 0)
```
Comment (aasif-patel) [inline erpnext/stock/report/available_batch_report/test_available_batch_report.py:43]: Are we expecting 0 result from the report execution?

File: erpnext/stock/report/available_batch_report/test_available_batch_report.py [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+		filters = self.make_filters(show_item_name=False)
+		columns = available_batch_report.get_columns(filters)
+		fieldnames = [col["fieldname"] for col in columns]
+		self.assertNotIn("item_name", fieldnames)
+
+	def test_to_date_today_with_expiry_check(self):
+		# Ensure the batch has a valid expiry date in the future
+		self.batch.expiry_date = today()
+		self.batch.reload()
+		self.batch.save()
```
Comment (aasif-patel) [inline erpnext/stock/report/available_batch_report/test_available_batch_report.py:140]: wrong use of reload(), you need to save the batch instance after updating it, then only reload it

File: erpnext/stock/report/available_batch_report/test_available_batch_report.py [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+
+		filters = self.make_filters(to_date=today(), item_code=self.item.name)
+		data = available_batch_report.get_data(filters)
+		self.assertIsInstance(data, list)
```
Comment (aasif-patel) [inline erpnext/stock/report/available_batch_report/test_available_batch_report.py:144]: checking type does make sense here, as get_data always return list,

File: erpnext/stock/report/available_batch_report/test_available_batch_report.py [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+
+	def test_to_date_today_with_expired_batches_included(self):
+		# Expired batch with include_expired_batches=True
+		self.batch.expiry_date = "2000-01-01"
+		self.batch.reload()
+		self.batch.save()
+
+		filters = self.make_filters(to_date=today(), item_code=self.item.name, include_expired_batches=True)
+		data = available_batch_report.get_data(filters)
+		self.assertIsInstance(data, list)
```
Comment (aasif-patel) [inline erpnext/stock/report/available_batch_report/test_available_batch_report.py:154]: same changes as above

File: erpnext/stock/report/available_batch_report/test_available_batch_report.py [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+				{"doctype": "Warehouse Type", "name": "Raw Material", "warehouse_type": "Raw Material"}
+			).insert()
+		warehouse_type = "Raw Material"
+		self.warehouse.warehouse_type = warehouse_type
+		self.warehouse.is_group = 0
+		self.warehouse.save()
+
+		filters = self.make_filters(warehouse_type=warehouse_type)
+		data = available_batch_report.get_data(filters)
+		self.assertIsInstance(data, list)
```
Comment (aasif-patel) [inline erpnext/stock/report/available_batch_report/test_available_batch_report.py:169]: same it always return list

**Review by aasif-patel — CHANGES_REQUESTED** [review]

**Comment by aasif-patel** [comment]
@ashishverma8848 what is the update on this
CC: @pallavi238848

**Comment by ashishverma8848** [comment]
> @ashishverma8848 what is the update on this CC: @pallavi238848

Worked on it. Pushed on vinod's repo to see what is the new percentage. if the percentage i'll get above 90 i'll push it today

**Comment by aasif-patel** [comment]
Another PR https://github.com/8848digital/erpnext/pull/2659 is raised for this changes hence closing
